### PR TITLE
Internal: Update PR template

### PR DESCRIPTION
**Pull request by ayeshakmaz**
Update the PR template (and fix a typo)

### CODE REVIEW

File: .github/pull_request_template.md [hunk 1]
```diff
@@ -1,18 +1,20 @@
...
-* Is it accessible?
-* Is it documented?
-* Have you involved other stakeholders (such as a Pinterest Designer)?
--->
+## Test Plan
+
+- [ ] Added Unit Tests
+- [ ] Added Accessibility Tests
+- [ ] Visually checked for regressions
+- [ ] Stakeholder feedback (ex. Gestalt designers)
```
Comment (AlbertCarreras) [inline .github/pull_request_template.md:20]: _[GOOD] Accessibility: validate keyboard & screen reader interaction
_[GOOD] Browser Support: test in multiple browsers (Chrome / Safari / Firefox)
_[GOOD] Dark Mode: view your component in dark mode
_[GOOD] Documentation: document your component & provide examples
_[GOOD] Responsiveness: view your component in sm/md/lg screen sizes
_[GOOD] RTL: Right to Left support
_[GOOD] Flow Tests
_[GOOD] Unit Tests

Is it worth incorporating any of the ones in the TDD in here? for instance...

[  ] Added/Updated Documentation
[  ] Checked for RTL support
etc.

File: .github/pull_request_template.md [hunk 2]
```diff
@@ -1,18 +1,20 @@
...
-* Is it tested?
-* Is it accessible?
-* Is it documented?
-* Have you involved other stakeholders (such as a Pinterest Designer)?
--->
+## Test Plan
+
+- [ ] Added Unit Tests
+- [ ] Added Accessibility Tests
+- [ ] Visually checked for regressions
```
Comment (AlbertCarreras) [inline .github/pull_request_template.md:19]: What does this mean?

**Review by AlbertCarreras — APPROVED** [review]
LGTM. Added a couple of comments for you to consider. If not, looking good!

File: .github/pull_request_template.md (hunk 2 above)
Comment (ayeshakmaz) [reply .github/pull_request_template.md]: Ended up removing it, but it was a reminder to look at related components when making a "breaking" change

File: .github/pull_request_template.md [hunk 3]
```diff
@@ -1,18 +1,20 @@
...
 <!--
 What is the purpose of this PR?
 
-* What is the context surrounding this PR? Include links if possible.
-* What kind of feedback do you want?
 * Have you [formatted the PR title](https://github.com/pinterest/gestalt/#releasing)? `ComponentName: Description`
 -->
 
-## Test Plan
+## Links
```
Comment (dangerismycat) [inline .github/pull_request_template.md:9]: nit: can we make these `###`? `##` seems jarringly large

Comment (ayeshakmaz) [reply .github/pull_request_template.md:9]: against my better a11y judgement, yes :P

File: .github/pull_request_template.md [hunk 4]
```diff
@@ -1,18 +1,21 @@
+# Description
```
Comment (dangerismycat) [inline .github/pull_request_template.md:1]: and maybe `##` for this one? (sorry to nitpick 😬)

**Review by dangerismycat — APPROVED** [review]
🚢🚢🚢🚢🚢

**Pull request merged**